### ISSUE TITLE
Checkbox accessibility update

### ISSUE THREAD

**Issue by cathystoli** [chanzuckerberg/sci-components#690]
**Issue:** 
1. Cannot check or uncheck the checkbox component using a keyboard
 
2. No focus control indicator

**Solution:**
1. Allow the component to be utilized via keyboard controls

2. Add focus control indicator to component

https://sds.czi.design/009eaf17b/p/548a3d-control-inputs/t/70a4ba
